### DISCUSSION TITLE
-quality and -compress Group4

### DISCUSSION THREAD

**Discussion by pbl987** [ImageMagick/ImageMagick#5692] [Help]
There is no difference in in file size for these command:

`magick .\Input.pdf -monochrome -compress Group4 **-quality 100** output%d.tiff`

`magick .\Input.pdf -monochrome -compress Group4 **-quality 10** output%d.tiff`

`magick .\Input.pdf -monochrome **-quality 10** -compress Group4 output%d.tiff`

`Version: ImageMagick 7.1.0-51 Q16-HDRI x64 aea87b5:20221016`

Is this intended?

**Comment by fmw42** [comment]
Default tiff output is compress none. So -quality will have no effect.  TIFF supports compress of none, LZW, ZIP and JPG. I think only JPG supports -quality.

**Reply by pbl987** [reply]
Ok, but i am adding -compress Group4, which apparently does something.

[From the manual webpage](https://imagemagick.org/script/command-line-options.php#compress):
> [...]Use the [-quality](https://imagemagick.org/script/command-line-options.php#quality) option to set the compression level to be used by JPEG, PNG, MIFF, and MPEG encoders. Use the [-sampling-factor](https://imagemagick.org/script/command-line-options.php#sampling-factor) option to set the sampling factor to be used by JPEG, MPEG, and YUV encoders for down-sampling the chroma channels.

What would you suggest?

**Reply by fmw42** [reply]
Both -monochrome and group4 make the image binary

**Reply by pbl987** [reply]
I am sorry, what does that mean?

**Reply by fmw42** [reply]
I do not think -quality is relevant to uncompressed TIFF. If you use JPG compression in TIFF, then it would be.  Also after -monochrome, I do not think -group4 uses -quality either nor does -monochrome.  You have two options with -monochrome. The first is the do as you did and get dithered binary image. The second is to use -dither none before -monochrome and get no dithering. I am not sure that would make any difference in file size. But I would have to test that. Once the image is binary from -monochrome, I am not sure that group4 would do much besides set some values for faxing. But I am not an expert on group4 compression. 

So the only way to change the file size is to set the compression type ( -compression) for the TIFF output:  either none, LZW, ZIP or JPG.  If JPG you can then use -quality also before the output.

**Reply by pbl987** [reply]
Oh, Group4 is absolutely a compression. By the way, the original PDF just contains jpeg from a scanner.

 `> magick -density 300 '.\Input.pdf'  -density 150 -monochrome -compress Group4 output.tiff`
`> magick -density 300 '.\Input.pdf'  -density 150 -monochrome output2.tiff`
`> ls *.tiff`
  `  Directory: D:\Dateien\Dokumente\test`
`Mode                 LastWriteTime         Length Name`
`----                 -------------         ------ ----`
`-a---          23.10.2022    22:48          73910 output1.tiff`
`-a---          23.10.2022    22:48       17342252 output2.tiff `

**Reply by pbl987** [reply]
What i don't understand: where in the documentation is it described what to do regarding TIFF compression? E.g. i would like to look at possible options for LZW and ZIP, but can't find anything on that topic.

**Reply by Jmuccigr** [reply]
My guess is that once you've set Group4 compression, there's no quality to worry about. It's a lossless compression of a binary image. It doesn't offer differing quality options (as jpeg does).

**Reply by pbl987** [reply]
@Jmuccigr I think that too.

@fmw42 
> So the only way to change the file size is to set the compression type ( -compression) for the TIFF output: either none, LZW, ZIP or JPG. If JPG you can then use -quality also before the output.

Obviously i can't do that, as -compress is set to Group4. (also there is no -compression ?). I am curious what you have in mind, maybe we can play with TIFF options a bit.

Kind regards.

**Comment by snibgo** [comment]
> E.g. i would like to look at possible options for LZW and ZIP, but can't find anything on that topic.

As far as I know, IM has no options specific to LZW or ZIP compression.

**Comment by snibgo** [comment]
@pbl987: What are your goals? Do you have a a problem that needs solving?

You may find that "-compress Zip" makes smaller TIFF files than "-compress GRoup4", if smallest size is your goal.
